src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/LanguageServices/Handlers/CommandLineNotificationHandler.cs

```diff
@@ -1,38 +1,32 @@
+namespace Microsoft.VisualStudio.ProjectSystem.LanguageServices.Handlers;
+
+/// <summary>
+///     An indirection that sends design-time build results in the form of command-line arguments to the F# language-service.
+/// </summary>
+/// <remarks>
+///     This indirection is needed because Microsoft.VisualStudio.ProjectSystem.FSharp does not have InternalsVisibleTo access to Roslyn.
+/// </remarks>
+[Export(typeof(IWorkspaceUpdateHandler))]
+[method: ImportingConstructor]
+internal class CommandLineNotificationHandler(UnconfiguredProject project) : IWorkspaceUpdateHandler, ICommandLineHandler
 {
-    /// <summary>
-    ///     An indirection that sends design-time build results in the form of command-line arguments to the F# language-service.
-    /// </summary>
     /// <remarks>
-    ///     This indirection is needed because Microsoft.VisualStudio.ProjectSystem.FSharp does not have InternalsVisibleTo access to Roslyn.
+    /// See <see cref="FSharpCommandLineParserService.HandleCommandLineNotifications"/> for an example of this export.
     /// </remarks>
-    [Export(typeof(IWorkspaceUpdateHandler))]
-    internal class CommandLineNotificationHandler : IWorkspaceUpdateHandler, ICommandLineHandler
-    {
-        [ImportingConstructor]
-        public CommandLineNotificationHandler(UnconfiguredProject project)
-        {
-            // See FSharpCommandLineParserService.HandleCommandLineNotifications for an example of this export
-            CommandLineNotifications = new OrderPrecedenceImportCollection<Action<string?, BuildOptions, BuildOptions>>(projectCapabilityCheckProvider: project);
-        }
-
-        /// <remarks>
-        /// See <see cref="FSharpCommandLineParserService.HandleCommandLineNotifications"/> for an export.
-        /// </remarks>
-        [ImportMany]
-        public OrderPrecedenceImportCollection<Action<string?, BuildOptions, BuildOptions>> CommandLineNotifications { get; }
+    [ImportMany]
+    public OrderPrecedenceImportCollection<Action<string?, BuildOptions, BuildOptions>> CommandLineNotifications { get; } = new(projectCapabilityCheckProvider: project);
 
-        public void Handle(IWorkspaceProjectContext context, IComparable version, BuildOptions added, BuildOptions removed, ContextState state, IManagedProjectDiagnosticOutputService logger)
+    public void Handle(IWorkspaceProjectContext context, IComparable version, BuildOptions added, BuildOptions removed, ContextState state, IManagedProjectDiagnosticOutputService logger)
+    {
+        foreach (Action<string?, BuildOptions, BuildOptions> value in CommandLineNotifications.ExtensionValues())
         {
-            foreach (Lazy<Action<string?, BuildOptions, BuildOptions>, IOrderPrecedenceMetadataView> value in CommandLineNotifications)
-            {
-                value.Value(context.BinOutputPath, added, removed);
-            }
+            value(context.BinOutputPath, added, removed);
```

Comment:
This [comment]( https://github.com/dotnet/project-system/pull/9605#discussion_r1861717158) should come to here, it is outdated in original place